src/databricks/labs/ucx/source_code/python_libraries.py

```diff
@@ -71,12 +72,16 @@ def _install_library(self, path_lookup: PathLookup, library: Path) -> MaybeDepen
         # invoke pip install via subprocess to install this library into self._lib_install_folder
         venv = self._temporary_virtual_environment(path_lookup).as_posix()
         existing_packages = os.listdir(venv)
-        pip_install_arguments = ["pip", "install", library.name, "-t", venv]
-        try:
-            subprocess.run(pip_install_arguments, check=True)
-        except CalledProcessError as e:
-            problem = DependencyProblem("library-install-failed", f"Failed to install {library}: {e}")
-            return MaybeDependency(None, [problem])
+        if library.suffix == ".egg":
+            with zipfile.ZipFile(library, "r") as zip_ref:
+                zip_ref.extractall(venv)  # There is a risk of overwritting existing files here!
```

Comment:
I am not able to use `easy_install` as the [setuptools documentation mentions](https://setuptools.pypa.io/en/latest/deprecated/python_eggs.html). `easy_install` can not be found anymore, which is a bit odd as the setuptools change log mention it is deprecated in [v58.3.0](https://setuptools.pypa.io/en/stable/history.html#v58-3-0), which I tried and no luck.

I tried in Databricks as well, but no `easy_install` there either, while up to runtime 14.0 [Python eggs are supported](https://learn.microsoft.com/en-us/azure/databricks/archive/legacy/workspace-libraries#--upload-a-jar-python-egg-or-python-wheel-file).

<img width="669" alt="Screenshot 2024-05-29 at 11 13 43" src="https://github.com/databrickslabs/ucx/assets/5946784/b331e2e9-581a-4b9b-9e2c-6aea564dc5ba">

Another route would be to use the `setup.py` from the (unzipped) egg. However, this is not always present, for example, the one I am using for testing.

The unzipping does let the integration test pass, though, installing would be a preferred route